src/sidebar/config/build-settings.js

```diff
@@ -178,55 +130,78 @@ async function fetchGroupsAsync(config, rpcCall) {
+ * - From a known parent frame, requested by RPC (e.g. LMS).
+ *   Send a single RPC request to a known frame to retrieve host configuration
+ * - From an ancestor frame, using RPC (deprecated)
+ *   Send RPC requests for configuration to ancestor frames until we
+ *   (hopefully) find a host configuration.
  *
- * @param {SidebarConfig} appConfig
- * @param {Window} window_ - Test seam.
- * @return {Promise<MergedConfig>} - The merged settings.
+ * In cases where configuration is retrieved by RPC, the `configFromURL` object
+ * values are not retained in the returned host configuration. Note the
+ * exception related to the `group` property below.
+ *
+ * @param {HostConfig} configFromURL
+ * @param {Window} [window_]
+ * @returns {Promise<HostConfig>}
  */
-export async function fetchConfig(appConfig, window_ = window) {
-  const hostConfig = hostPageConfig(window);
-
-  const requestConfigFromFrame = hostConfig.requestConfigFromFrame;
+async function findHostConfiguration(configFromURL, window_) {
+  const requestConfigFromFrame = configFromURL.requestConfigFromFrame;
 
   if (!requestConfigFromFrame) {
-    // Directly embed: just get the config.
-    return fetchConfigEmbed(appConfig, hostConfig);
+    // In this case, the `configFromURL` is retained and used as host configuration
+    return configFromURL;
   }
+
   if (typeof requestConfigFromFrame === 'string') {
-    // Legacy: send RPC to all parents to find config. (deprecated)
-    // nb. Browsers may display errors in the console when messages are sent to frames
-    // that don't match the origin filter".
-    return await fetchConfigLegacy(appConfig, window_);
+    return await fetchConfigFromAncestorFrame(requestConfigFromFrame, window_);
   } else if (
     typeof requestConfigFromFrame.ancestorLevel === 'number' &&
     typeof requestConfigFromFrame.origin === 'string'
   ) {
-    // Know parent frame: send RPC directly to the parent.
-    const parentFrame = getAncestorFrame(
-      requestConfigFromFrame.ancestorLevel,
-      window_
-    );
-
-    const rpcMergedConfig = await fetchConfigRpc(
-      appConfig,
-      parentFrame,
+    const hostConfiguration = await fetchHostConfigRPC(
+      getAncestorFrame(requestConfigFromFrame.ancestorLevel, window_),
       requestConfigFromFrame.origin
     );
-    // Add back the optional focused group id from the host page config
-    // as this is needed in the Notebook.
-    return {
-      ...rpcMergedConfig,
-      ...(hostConfig.group ? { group: hostConfig.group } : {}),
-    };
+
+    // Manually add a `group` property to the host configuration if
+    // it was present on `configFromURL`. This allows a focused group to be
+    // available to the Notebook within LMS contexts.
+    // FIXME: This could be organized/typed more elegantly.
+    if (configFromURL.group && !hostConfiguration.group) {
+      hostConfiguration.group = configFromURL.group;
+    }
+    return hostConfiguration;
```

Comment:
This is not very nice at present...I wanted to keep changes limited but am open to how we might better handle this.

Reply:
We've now got the most of the messaging infrastructure needed to enable the sidebar to communicate directly with the notebook. This infrastructure allows for sending messages even before the app receiving them has finished loading. We could in future use this for communicating the group to the notebook when it initially loads and when it is subsequently changed in the sidebar. I think Eduardo wrote some draft code for this somewhere, so we could try to dig that up.